### PR TITLE
feat: added krishnamurthy number verificator

### PR DESCRIPTION
**Pull request by jspmic**
#### Description of Change
- Created an implementation of the krishnamurthy number verificator
- Added test cases

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: math/krishnamurthy.c [hunk 1]
```diff
@@ -0,0 +1,76 @@
+#include <stdio.h> // For I/O ops
+#include <assert.h> //For test cases
```
Comment (realstealthninja) [inline math/krishnamurthy.c]:
```suggestion
#include <stdio.h> /// For I/O operations
#include <assert.h> /// For test cases
#include <stdbool.h> /// For boolean
```

File: math/krishnamurthy.c [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+*/
+
+
+/**
+ * @description n! = n*(n-1)*(n-2)*...*1
+ * @brief calculates the factorial of a number
+ * @param n the number we want to calculate the factorial
+ * @return an integer representing the factorial of n
+ */
+int factorial(int n){
```
Comment (realstealthninja) [inline math/krishnamurthy.c:19]: there are implementations of factorial available in this repository you could use that if you'd like

File: math/krishnamurthy.c [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+	sum += factorial(n%10);
+	return get_digit(div,sum);
+    }
+}
+
+/*
+ * @brief Verifies if a given number
+ * is a krishnamurthy number
+ * @param n the number to be verified as a krishnamurthy number
+ * @returns 1 if n is a krishnamurthy number
```
Comment (realstealthninja) [inline math/krishnamurthy.c]: you should use booleans here

File: math/krishnamurthy.c [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+ * @returns 0 if n is not a krishnamurthy number
+ */
+int krishnamurthy(int n){
```
Comment (realstealthninja) [inline math/krishnamurthy.c]:
```suggestion
bool krishnamurthy(int n){
```

File: math/krishnamurthy.c [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+/**
+ * @brief Implementation of test cases
+ * @return void
+ */
+static void tests(){
+    assert(krishnamurthy(1) == 1);
+    assert(krishnamurthy(13) == 0);
+    assert(krishnamurthy(45) == 0);
+    assert(krishnamurthy(145) == 1);
+    assert(krishnamurthy(239) == 0);
```
Comment (realstealthninja) [inline math/krishnamurthy.c]:
```suggestion
    assert(krishnamurthy(1) == true);
    assert(krishnamurthy(13) == false);
    assert(krishnamurthy(45) == false);
    assert(krishnamurthy(145) == true);
    assert(krishnamurthy(239) == false);
```

File: math/krishnamurthy.c [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    printf("All tests were successful!\n");
+}
+
```
Comment (realstealthninja) [inline math/krishnamurthy.c:65]:
```suggestion

/**
 * @brief main function
 * @returns 0 on successful exit 
 */
```

File: math/krishnamurthy.c [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    return n*factorial(n-1);
+}
+
+/**
+ * @brief Calculates the sum of the factorial of each digit
+ * @param n the number to calculate the sum of the factorial of its digits
+ * @param sum the sum variable to be incremented 
+ * @returns an integer representing that sum
+ */
+int get_digit(int n, int sum){
```
Comment (realstealthninja) [inline math/krishnamurthy.c]: this gets sum not digit

File: math/krishnamurthy.c [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+/*
+ * @brief Verifies if a given number
+ * is a krishnamurthy number
+ * @param n the number to be verified as a krishnamurthy number
+ * @returns 1 if n is a krishnamurthy number
+ * @returns 0 if n is not a krishnamurthy number
+ */
+int krishnamurthy(int n){
+    int sum=0;
```
Comment (realstealthninja) [inline math/krishnamurthy.c]:
```suggestion
```
unnecessary variable should be locally defined

File: math/krishnamurthy.c [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+    return get_digit(n,sum)==n;
```
Comment (realstealthninja) [inline math/krishnamurthy.c]:
```suggestion
    return get_digit(n)==n;
```

File: math/krishnamurthy.c [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+
+/**
+ * @brief Calculates the sum of the factorial of each digit
+ * @param n the number to calculate the sum of the factorial of its digits
+ * @param sum the sum variable to be incremented 
+ * @returns an integer representing that sum
+ */
+int get_digit(int n, int sum){
+
+    if (n<10){
```
Comment (realstealthninja) [inline math/krishnamurthy.c]:
```suggestion
    int sum = 0;
    if (n<10){
```

File: math/krishnamurthy.c (hunk 7 above)
Comment (realstealthninja) [inline math/krishnamurthy.c]:
```suggestion
int get_digit(int n){
```
sum should be locally defined

**Review by realstealthninja — CHANGES_REQUESTED** [review]
Thank you for your contributions please refer to the contribution guidelines to fix you documentation format of source code.
every file should be documented with a heading multi-line comment 
containing references and descriptions of the given algorithm

File: math/krishnamurthy.c (hunk 2 above)
Comment (jspmic) [reply math/krishnamurthy.c:19]: i didn't find any satisfying implementation of the factorial, so i made my own! The implementations already established were not implemented using a functional approach!!!

File: math/krishnamurthy.c (hunk 3 above)
Comment (jspmic) [reply math/krishnamurthy.c]: thanks for teaching me how to include booleans, i didn't know... sorry!

File: math/krishnamurthy.c (hunk 7 above)
Comment (jspmic) [reply math/krishnamurthy.c]: i even found a solution... i didn't even need sum!! thanks for your feedback, i learned something for free!! Thanks

**Pull request closed without merging**